var/spack/repos/builtin/packages/py-matplotlib/package.py

```diff
@@ -69,22 +70,45 @@ class PyMatplotlib(PythonPackage):
     # https://matplotlib.org/tutorials/introductory/usage.html#backends
     # From `lib/matplotlib/rcsetup.py`:
     interactive_bk = [
+        # https://github.com/spack/spack/issues/32694
+        "macosx",
+        # GTK
+        conditional("gtk", when="@:2"),
+        conditional("gtkagg", when="@:2"),
+        conditional("gtkcairo", when="@:2"),
         "gtk3agg",
         "gtk3cairo",
-        "macosx",
-        "nbagg",
-        "qt4agg",
-        "qt4cairo",
+        conditional("gtk4agg", when="@3.5:"),
+        conditional("gtk4cairo", when="@3.5:"),
+        # Qt
+        conditional("qtagg", when="@3.5:"),
+        conditional("qtcairo", when="@3.5:"),
```

Comment:
From what I can tell, the intention of having `qtagg` and `qtcairo` is to have a single backend for all Qt versions. The user then specifies the `QT_API` environment variable in order to specify which backend they want. See https://github.com/matplotlib/matplotlib/pull/19255. However, we have separate packages for `py-pyqt4/5/6` and `py-pyside4/5/6`, so there's no way to have a single dependency. Instead, I'm hardcoding to use `py-pyqt6`, but that's not really the intent of this backend.